### PR TITLE
[WIP] Reorg the Node and Build secrets files

### PR DESCRIPTION
**Pull request by mburke5678**
In researching https://bugzilla.redhat.com/show_bug.cgi?id=2093995, I noticed that there is redundant information about Secrets in a couple of different files:

This PR reorganizes the Node and Build _secret_ files:

* Replaced the text in CICD -> Builds -> Creating build inputs -> **[What is a secret?](https://46882--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/creating-build-inputs#nodes-pods-secrets-about_creating-build-inputs)** with the text in **[Understanding secrets](https://docs.openshift.com/container-platform/4.12/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-about_nodes-pods-secrets)** from Nodes.  

* Added CICD -> Builds -> Creating build inputs -> [**Types of secrets**](https://46882--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/creating-build-inputs#nodes-pods-secrets-about-types_creating-build-inputs) from Nodes. 
 
* Replaced the text in CICD -> Builds -> Creating build inputs -> [Updates to secrets](https://46882--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/creating-build-inputs#nodes-pods-secrets-updating_creating-build-inputs) with the text in **[Understanding how to update secrets](https://docs.openshift.com/container-platform/4.12/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-updating_nodes-pods-secrets)** from Nodes.

* Added a link after  CICD -> Builds -> Creating build inputs -> [**Creating secrets**](https://46882--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/creating-build-inputs#builds-creating-secrets_creating-build-inputs) in Builds to **Creating secrets** in Node that shows how to configure all secret types.

* Combined **Using secrets** from  CICD -> Builds -> Creating build inputs ([current docs](https://46882--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/creating-build-inputs#builds-creating-secrets_creating-build-inputs)) and Nodes docs ([current docs](https://46882--docspreview.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets#nodes-pods-secrets-creating_nodes-pods-secrets)). Using the same module in both assemblies: [Creating build inputs](https://46882--docspreview.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets#nodes-pods-secrets-creating_nodes-pods-secrets) and [Providing sensitive data to pods by using secrets](https://46882--docspreview.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets#nodes-pods-secrets-creating_nodes-pods-secrets).

* Added example secret configurations to  CICD -> Builds -> Creating build inputs -> [**Using secrets**](https://46882--docspreview.netlify.app/openshift-enterprise/latest/cicd/builds/creating-build-inputs#builds-using-secrets_creating-build-inputs) from [Understanding how to create secret](https://docs.openshift.com/container-platform/4.13/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-creating_nodes-pods-secrets)..

**Questions:**
Should [Secret data keys](https://docs.openshift.com/container-platform/4.10/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-about-keys_nodes-pods-secrets) be in builds?
What is the list of types of secrets? The [Builds module has 6 types](https://docs.openshift.com/container-platform/4.10/cicd/builds/creating-build-inputs.html#builds-secrets-overview-types_creating-build-inputs); [Nodes has 4](https://docs.openshift.com/container-platform/4.10/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-about-types_nodes-pods-secrets) (The `dockercfg` and `dockerconfigjson` types are missing).

### CODE REVIEW

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: modules/nodes-pods-secrets-creating-basic.adoc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+* password: the password or token for authentication
+
+[NOTE]
+====
+You can use the `stringData` parameter to use clear text content.
+====
+
+The Kubernetes API verifies that the required keys are set for a Secret of this type.
+
+You can create an opaque secret to pass the credentials used for basic authentication. Using the basci authentication secret helps better defines the purpose of the secret, and sets a convention for what key names to expect.
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/nodes-pods-secrets-creating-basic.adoc:21]: Thank you! I should have said I haven't proofread it all yet. Just wanted a sense that I am going in the right direction.

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by kalexand-rh** [comment]
Only versions 4.8+ are still in maintenance. I am removing labels from earlier versions from this PR.

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.13` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.12. And any PR going into `main` must also target the latest version branch (enterprise-4.13).

If the update in your PR does NOT apply to version 4.13 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by kalexand-rh** [comment]
Only versions 4.10+ are still in maintenance. I am removing labels from earlier versions from this PR.

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.14` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.13. And any PR going into `main` must also target the latest version branch (enterprise-4.14).

If the update in your PR does NOT apply to version 4.14 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by kalexand-rh** [comment]
Only versions 4.11+ are still in maintenance. I am removing labels from earlier versions from this PR.

**Comment by kalexand-rh** [comment]
/remove-lifecycle stale

@mburke5678, is this still valid?

**Comment by mburke5678** [comment]
@kalexand-rh WDYT? Are these changes worthwhile to persue?

**Comment by kalexand-rh** [comment]
> @kalexand-rh WDYT? Are these changes worthwhile to persue?

I think it's worth rebasing. If that works, it'll be worth putting out for QE. I think the reorg makes sense, and node content has a substantial number of views.

**Pull request closed without merging**